src/controllers/userControllers.ts

```diff
@@ -56,13 +59,17 @@ export const userLogin = async (req: Request, res: Response) => {
     } else {
       // @ts-ignore
       if (user.userRole.name === "seller") {  
-        const token = Math.floor(Math.random() * 90000 + 10000);
-        //@ts-ignore
-        await Token.create({ token: token, userId: user.id });
-        await mailService.sendEmailService(user, SUBJECTS.CONFIRM_2FA, verifyOtpTemplate(token), token);
+        const otp = generateRandomNumber();
+        const token = await generateMagicLinkToken(otp, user);
+        const link =
+          process.env.NODE_ENV !== "production"
+            ? `http://localhost:${env.port}/api/v1/users/2fa-verify/${token}`
+            : `https://eagles-ec-be-development.onrender.com/api/v1/users/2fa-verify/${token}`;
```

Comment:
don't hardcode this things, create .env for url, this this will be called from front-end

Reply:
yeap